### ISSUE TITLE
Clob columns defined with unique constraints do not work

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#254]
```
Steps to reproduce problem.
CREATE TABLE MYTABLE (MYCOL text not null unique)
INSERT INTO MYTABLE(MYCOL) VALUES('THIS IS A TEST')
INSERT INTO MYTABLE(MYCOL) VALUES('THIS IS NOT THE SAME') <- This fails

Expected:
Insert to succeed

Actual:
Fails with unique constraint violation exception;

H2 - 1.3.146
Linux - Ubuntu 10.10
Java - 1.6.0_21 64 bit
Filesystem - ext3

Right now we use postgres as our database, but we are looking at embedded 
database solutions to run things like demos. I've noticed a few problems with 
the postgres data type 'text' as a Clob. This is only one of them.

```

Original issue reported on code.google.com by `sjoy...@integrityemr.com` on 10 Nov 2010 at 2:01

**Comment by GoogleCodeExporter** [comment]
```
When I tried it:
- INSERTs succeed under version 1.2.145
- Second INSERT fails under trunk revision 3092 (which I believe is essentially 
the same as 1.3.146 beta)

Since 1.3.x is in beta, I'd suggest using 1.2.145 instead.

---

Version 1.2.145:

CREATE TABLE MYTABLE (MYCOL text not null unique);
Update count: 0
(1 ms)

INSERT INTO MYTABLE(MYCOL) VALUES('THIS IS A TEST');
Update count: 1
(1 ms)

INSERT INTO MYTABLE(MYCOL) VALUES('THIS IS NOT THE SAME');
Update count: 1
(0 ms)

SELECT * FROM MYTABLE;
MYCOL  
THIS IS A TEST
THIS IS NOT THE SAME
(2 rows, 1 ms)

---

Trunk revision 3092:

CREATE TABLE MYTABLE (MYCOL text not null unique);
Update count: 0
(2 ms)

INSERT INTO MYTABLE(MYCOL) VALUES('THIS IS A TEST');
Update count: 1
(1 ms)

INSERT INTO MYTABLE(MYCOL) VALUES('THIS IS NOT THE SAME');
Unique index or primary key violation: "CONSTRAINT_INDEX_8 ON 
PUBLIC.MYTABLE(MYCOL)"; SQL statement:
INSERT INTO MYTABLE(MYCOL) VALUES('THIS IS NOT THE SAME') [23001-146] 
23001/23001 (Help)

SELECT * FROM MYTABLE;
MYCOL  
THIS IS A TEST
(1 row, 1 ms)
```

Original comment by `eric.anl...@gmail.com` on 10 Nov 2010 at 4:02

**Comment by GoogleCodeExporter** [comment]
```
Thanks for your help! Yes, this is a bug, it is now fixed in the trunk 
(revision 3094).
```

Original comment by `thomas.t...@gmail.com` on 11 Nov 2010 at 5:14
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in version 1.2.147 and newer.
```

Original comment by `thomas.t...@gmail.com` on 12 Dec 2010 at 12:36
- Changed state: **Fixed**
